### PR TITLE
Fix time-based index querying

### PR DESCRIPTION
**Pull request by pitr**
- remove order by from index read
- fix time bucket bounding bug
- replace IN with futures for time bucket AND for split tag value

### CODE REVIEW

File: src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java [hunk 1]
```diff
@@ -843,17 +855,27 @@ private static void logQuery(DatastoreMetricQuery query,
...
         final List<DataPointsRowKey> rowKeys = new LinkedList<>();
-        collectFromRowKeySplitIndex(rowKeys,
-                query,
-                useSplitField,
-                useSplit,
-                query.getStartTime(),
-                query.getEndTime(),
-                limit);
+        if (m_cassandraConfiguration.isUseTimeIndexRead()) {
+            List<ResultSetFuture> futures = calculateReadTimeBuckets(query.getStartTime(), query.getEndTime())
```
Comment (alexkorotkikh) [inline src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java]: This `if` branch looks very similar to [that one](https://github.com/zalando-zmon/kairosdb/pull/114/files#diff-5cc760fcd7185d42167a9ce3ff854de4R839-R846). Can we probably try to generalize it and extract to separate method, e.g. `collectFromRowTimeKeySplitIndex`?

Comment (pitr) [reply src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java]: fixed

File: src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java [hunk 2]
```diff
@@ -630,16 +624,10 @@ public long calculateRowTimeWrite(long timestamp) {
         return (timestamp - (Math.abs(timestamp) % m_rowWidthWrite));
     }
 
-    private List<Long> calculateReadTimeBuckets(long startTime, long endTime) {
-        LinkedList<Long> buckets = new LinkedList<Long>();
+    private List<Long> calculateReadTimeBuckets(long startTimeBucket, long endTimeBucket) {
```
Comment (alexkorotkikh) [inline src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java:627]: I'd just return `IntStream.range(startTimeBucket, endTimeBucket)` from this method

**Review by alexkorotkikh — APPROVED** [review]

File: src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java (hunk 2 above)
Comment (pitr) [reply src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java:627]: we have a step size (not just `i++`)

**Comment by pitr** [comment]
:+1:

**Comment by a1exsh** [comment]
:+1:

**Pull request merged**